mne/utils/check.py

```diff
@@ -461,12 +458,13 @@ def _check_option(parameter, value, allowed_values, error=ValueError):
         return True
 
     # Prepare a nice error message for the user
-    msg = ("Invalid value for the '{parameter}' parameter. "
-           "{options}, but got '{value}' instead.")
+    msg = ('Invalid value for the {parameter} parameter. '
+           '{options}, but got {value!r} instead.')
     if len(allowed_values) == 1:
-        options = "The only allowed value is '%s'" % allowed_values[0]
+        options = 'The only allowed value is %r' % allowed_values[0]
     else:
-        options = "Allowed values are "
-        options += ", ".join(["'%s'" % v for v in allowed_values[:-1]])
-        options += " and '%s'" % allowed_values[-1]
-    raise error(msg.format(parameter=parameter, options=options, value=value))
+        options = 'Allowed values are '
+        options += ', '.join(['%r' % v for v in allowed_values[:-1]])
+        options += ' and %r' % allowed_values[-1]
+    raise ValueError(msg.format(parameter=parameter, options=options,
```

Comment:
are you sure that it will always be a `ValueError`? I'm fine with the change and change it back if a TypeError appears.

Reply:
Scratch that. Actually to check if an option is a subclass of the `allowed_values` the function should be quite different: 

```py
def _check_option(option_name, option_val, allowed_values, type_of_check='one of', ...):
```

and allow for `type_of_check='sublcass'`, etc..

Reply:
Why would I want to check whether an option is a subclass of `allowed_values`?

I've gone through the entire codebase now. This function only has to deal with `str`, `None`, `True` and `False`. It also only needed to throw `ValueError`, hence I removed the code for throwing other types of errors.

The reason this function doesn't need to be more complicated than this, is that all it really does is form a very specific error message that just happens to be a very common one. When a different kind of check is required, the error message is likely to be different as well and you're better off not bending over just to use this function but just write the if-statement and `throw` yourself.

Reply:
I agree. Let’s not be too smart here. @wmvanvliet can you just see why CIs complain ?